src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,59 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_SIZE = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private int currentSize = INITIAL_SIZE;
+    private int size = 0;
+    private T[] table = createArray(currentSize);
+
+    private T[] createArray(int size) {
+        return (T[]) new Object[size];
+    }
 
+    public void push(T value) {
+        if (size >= currentSize) {
+            resize();
+        }
+        table[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        checkSize();
+        return table[size - 1];
     }
 
     public T pop() {
-        return null;
+        checkSize();
+        T value = peek();
+        table[size - 1] = null;
+        size--;
+        return value;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        currentSize = (int) (table.length * GROWTH_FACTOR);
+        System.arraycopy(table,0,table,0,currentSize);
```

Comment:
Pls make sure that your code is properly formatted (Ctrl+Alt+L (Windows/Linux) or⌥⌘L (macOS)).
Also it would be better to allocate a new array with the increased size, and then copy the existing elements into the new array.
`currentSize = (int) (table.length * GROWTH_FACTOR);
    T[] newTable = createArray(currentSize);
    System.arraycopy(table, 0, newTable, 0, size);
    table = newTable;`

Reply:
ok, done